cmake/partition_manager.cmake

```diff
@@ -6,29 +6,60 @@
 
 if(FIRST_BOILERPLATE_EXECUTION)
   get_property(
-    partition_manager_config_files
+    images
     GLOBAL PROPERTY
-    PARTITION_MANAGER_CONFIG_FILES
+    PM_IMAGES
     )
 
-  if(partition_manager_config_files)
+  if(images)
     # Partition manager is enabled because we have populated config
     # files.
 
+    set(generated_path include/generated)
+
+    foreach (image ${images})
+      get_property(build_dir GLOBAL PROPERTY ${image}PROJECT_BINARY_DIR)
+      set(generated_dir ${build_dir}/${generated_path})
+      list(APPEND partition_manager_config_file_dirs ${generated_dir})
+    endforeach()
+
     execute_process(
       COMMAND
       ${PYTHON_EXECUTABLE}
       ${ZEPHYR_BASE}/scripts/partition_manager.py
-      --input ${partition_manager_config_files}
-      --app-pm-config-dir ${PROJECT_BINARY_DIR}/include/generated
-      --app-build-dir ${PROJECT_BINARY_DIR}
+      --input-dirs ${partition_manager_config_file_dirs}
       RESULT_VARIABLE ret
       )
 
     if(ret EQUAL "1")
-      message( FATAL_ERROR "Partition Manager failed, aborting.")
+      message(FATAL_ERROR "Partition Manager failed, aborting.")
     endif()
 
+    foreach (image ${images})
```

Comment:
I don't see how this list is populated, but we need to make sure that the `app` image is inside it, so that the root app also get these output files. Alternatively we could have `foreach image ${images} + 'app'` or something like that.

Reply:
Fixed now.